components/sensor/dlms_meter.rst

```diff
@@ -0,0 +1,208 @@
+      baud_rate: 2400
+      rx_buffer_size: 1024 # Needed to receive the large packets send by the smart meter
+
+    dlms_meter:
+      decryption_key: "36C66639E48A8CA4D6BC8B282A793BBB" # change this to your decryption key!
+
+    sensor:
+      - platform: dlms_meter
+        voltage_l1:
+          name: "Voltage L1"
+        voltage_l2:
+          name: "Voltage L2"
+        voltage_l3:
+          name: "Voltage L3"
+        current_l1:
+          name: "Current L1"
+        current_l2:
+          name: "Current L2"
+        current_l3:
+          name: "Current L3"
+        active_power_plus:
+          name: "Active power taken from grid"
+        active_power_minus:
+          name: "Active power put into grid"
+        active_energy_plus:
+          name: "Active energy taken from grid"
+        active_energy_minus:
+          name: "Active energy put into grid"
+        reactive_energy_plus:
+          name: "Reactive energy taken from grid"
+        reactive_energy_minus:
+          name: "Reactive energy put into grid"
+
+    text_sensor:
+      - platform: dlms_meter
+        timestamp:
+          name: "timestamp"
+
+.. collapse:: Example yamls for special grid operators
+
+    NetzNoe / EVN
+
+    .. code-block:: yaml
+
+        # Sample configuration entry for netznow
+        uart:
+          rx_pin: GPIO4 # change this to the pin where you connected the mbus adapter board
+          baud_rate: 2400
+          rx_buffer_size: 1024 # Needed to receive the large packets send by the smart meter
+
+        dlms_meter:
+          decryption_key: "36C66639E48A8CA4D6BC8B282A793BBB" # change this to your decryption key!
+          provider: netznoe # (optional) key - only set if using evn
+
+        sensor:
+          - platform: dlms_meter
+            voltage_l1:
+              name: "Voltage L1"
+            voltage_l2:
+              name: "Voltage L2"
+            voltage_l3:
+              name: "Voltage L3"
+            current_l1:
+              name: "Current L1"
+            current_l2:
+              name: "Current L2"
+            current_l3:
+              name: "Current L3"
+            active_power_plus:
+              name: "Active power taken from grid"
+            active_power_minus:
+              name: "Active power put into grid"
+            active_energy_plus:
+              name: "Active energy taken from grid"
+            active_energy_minus:
+              name: "Active energy put into grid"
+            # EVN
+            power_factor:
+              name: "Power Factor"
+
+        text_sensor:
+          - platform: dlms_meter
+            timestamp:
+              name: "timestamp"
+            # EVN
+            meternumber:
+              name: "meterNumber"
+
+Configuration variables:
+
+- **decryption_key** (*Optional*, string, :ref:`templatable <config-templatable>`, 32 characters, case insensitive): The key to decrypt the
```

Comment:
isn't this required? or are there providers without encryption?

Reply:
Thanks, copy/paste error :)